### PR TITLE
[ONNX] Add a test file for _building.py

### LINKED ISSUE #138761: [ONNX] Test _building.py
We need a test to verify and guarantee the correct behavior on graph building.

### PR DESCRIPTION
**Pull request by titaiwangms**
Fixes #138761 

Add test file for _building.py to verify and guarantee the correct behavior on OpRecorder. Noted that the tests does not validate the model itself, but the expected behavior of the evaluator adding extra ops during input preprocessing.

### CODE REVIEW

File: test/onnx/exporter/test_building.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                    "": self.opset_version,
+                },
+                name="main_graph",
+            ),
+            ir_version=9,
+            producer_name="pytorch",
+            producer_version=torch.__version__,
+        )
+
+    def test_skippable_castlike(self):
```
Comment (justinchuby) [inline test/onnx/exporter/test_building.py:36]: I would refine the test name to explain a little better (e.g. "redundant castlike is omitted from graph" or something more succinct)

File: test/onnx/exporter/test_building.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        with onnxscript.evaluator.default_as(
+            tracer := self.recorder,
+        ):
+            cast = self.opset.CastLike(input_y, input_x)
+            _ = self.opset.Add(input_x, cast)
+
+        self.assertEqual(len(tracer.nodes), 1)
+        self.assertEqual(tracer.nodes[0].op_type, "Add")
+
+    def test_castlike_to_cast(self):
```
Comment (justinchuby) [inline test/onnx/exporter/test_building.py:52]:
```suggestion
    def test_castlike_to_cast_when<...>(self):
```

File: test/onnx/exporter/test_building.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            tracer := self.recorder,
+        ):
+            cast = self.opset.CastLike(input_y, input_x)
+            _ = self.opset.Add(input_x, cast)
+
+        self.assertEqual(len(tracer.nodes), 2)
+        self.assertEqual(tracer.nodes[0].op_type, "Cast")
+        self.assertEqual(tracer.nodes[1].op_type, "Add")
+
+    def test_process_python_constant(self):
```
Comment (justinchuby) [inline test/onnx/exporter/test_building.py:69]:
```suggestion
    def test_python_constant_added_as_constant_nodes(self):
```

File: test/onnx/exporter/test_building.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        input_x = _tensors.SymbolicTensor(
+            opset=self.opset, name="input_x", shape=ir.Shape([2, 3, 4])
+        )
+        new_shape = [3, 2, 4]
+
+        with onnxscript.evaluator.default_as(
+            tracer := self.recorder,
+        ):
+            _ = self.opset.Reshape(input_x, new_shape)
```
Comment (justinchuby) [inline test/onnx/exporter/test_building.py:78]: Should we test different types of constants?

**Review by justinchuby — APPROVED** [review]
Fantastic. Thanks for creating the tests! Some general comments on naming.

File: test/onnx/exporter/test_building.py (hunk 1 above)
Comment (titaiwangms) [reply test/onnx/exporter/test_building.py]: Done

File: test/onnx/exporter/test_building.py (hunk 4 above)
Comment (titaiwangms) [reply test/onnx/exporter/test_building.py:78]: I think for tracer tests, we only needs to take care of the tracer.nodes. Do you have a specific test case in mind? I guess you are referring to test on `_determine_input_dtype`, but I think the function is pretty straight forward though.

**Comment by titaiwangms** [comment]
I will address follow-ups if there is any (add more test if it's suggested.).

**Pull request closed without merging**